parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -265,6 +265,48 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Reset's the internal state of the light client.
+		///
+		/// `forked_at` - The hash of the header where the fork originated. The common ancestor.
+		///
+		/// This is only meant to be used in staging environments where long range forks
+		/// can happen.
+		///
+		/// Requires sudo user.
+		#[pallet::weight(1_000_000)]
+		#[transactional]
+		pub fn handle_long_range_fork(origin: OriginFor<T>, forked_at: H256) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let stored_header = <Headers<T>>::get(forked_at).ok_or(Error::<T>::MissingHeader)?;
+			let header = stored_header.header;
+
+			let required_descendants = T::DescendantsUntilFinalized::get() as usize;
+			let best_block_id = EthereumHeaderId { number: header.number, hash: forked_at };
+
+			// Return error if the fork occurs after the current finalized block
+			let current_finalized = <FinalizedBlock<T>>::get();
+			if best_block_id.number >= current_finalized.number {
+				return Err(Error::<T>::Unknown.into())
```

Comment:
Is an `Unknown` error sufficient because we only use this dispatchable during testing?

Reply:
Yeah so I also did not feel too great about overloading the `Unknown` error here, but this dispatchable is only going to be used in staging environments by one of us following a runbook to recover from a hard fork. So I did not want to pollute the interface by adding custom error messages for this one method.

Reply:
You can also just return an untyped string message instead of unknown:
```suggestion
				return Err("new best block is not valid".into())
```

That gets converted into `DispatchError::Other(some_string)`.